compiler/src/iree/compiler/Dialect/Util/IR/UtilTypes.h

```diff
@@ -255,6 +255,11 @@ using SetIntDivisibilityFn =
 // value. Returns None if the shape cannot be found.
 std::optional<ValueRange> findDynamicDims(Value shapedValue);
 
+// Walks the SSA use-def chain upwards to find the requested dimension of the
+// value if the dimension is dynamic. Returns the static size if the dim is
+// static, and null if the walk fails.
```

Comment:
maybe add a warning to this that it's inefficient - `If querying more than one dimension use findDynamicDims instead of calling this multiple times for the same shaped value.`